### PR TITLE
Add special methods for object

### PR DESCRIPTION
**Pull request by z33ky**
Adds special methods for `object` as per #773.

3.6 also added `__set_name__`, but I'm uncertain if simply creating the directory `stdlib/3.6` would suffice to add 3.6-specific type hints.

There are also still missing special methods, like `__prepare__`: I'm just confused to which class they belong.

### CODE REVIEW

File: stdlib/3/builtins.pyi [hunk 1]
```diff
@@ -38,9 +38,26 @@ class object:
     def __setattr__(self, name: str, value: Any) -> None: ...
     def __eq__(self, o: object) -> bool: ...
     def __ne__(self, o: object) -> bool: ...
+    def __lt__(self, o: object) -> bool: ...
```
Comment (JukkaL) [inline stdlib/3/builtins.pyi]: These 4 definitions will break mypy. It's best to leave them out and potentially file a separate issue about these.

File: stdlib/3/builtins.pyi [hunk 2]
```diff
@@ -38,9 +38,26 @@ class object:
...
+    def __le__(self, o: object) -> bool: ...
+    def __gt__(self, o: object) -> bool: ...
+    def __ge__(self, o: object) -> bool: ...
     def __str__(self) -> str: ...
     def __repr__(self) -> str: ...
     def __hash__(self) -> int: ...
+    def __del__(self) -> None: ...
```
Comment (JukkaL) [inline stdlib/3/builtins.pyi]: These don't seem to exist in Python 3.5.1: `__del__`, `__bytes__`, `__bool__`, `__getattr__`, `__get__`, `__set__`.

File: stdlib/2/__builtin__.pyi [hunk 3]
```diff
@@ -35,9 +35,23 @@ class object:
...
     def __eq__(self, o: object) -> bool: ...
     def __ne__(self, o: object) -> bool: ...
+    def __lt__(self, o: object) -> bool: ...
+    def __le__(self, o: object) -> bool: ...
+    def __gt__(self, o: object) -> bool: ...
+    def __ge__(self, o: object) -> bool: ...
     def __str__(self) -> str: ...
     def __repr__(self) -> str: ...
     def __hash__(self) -> int: ...
+    def __del__(self) -> None: ...
```
Comment (JukkaL) [inline stdlib/2/__builtin__.pyi]: This doesn't seem to exist.

File: stdlib/2/__builtin__.pyi [hunk 4]
```diff
@@ -35,9 +35,23 @@ class object:
...
+
+    def __getattr__(self, name: str) -> Any: ...
+    def __getattribute__(self, name: str) -> Any: ...
+    def __delattr__(self, name: str) -> None: ...
+    def __dir__(self) -> List[str]: ...
+
+    def __get__(self, instance: Optional[object], owner: Type) -> Any: ...
+    def __set__(self, instance: object, value: Any) -> None: ...
+    def __delete__(self, instance: object) -> None: ...
```
Comment (JukkaL) [inline stdlib/2/__builtin__.pyi]: These don't seem to exist: `__get__`, `__set__` and `__delete__`

**Comment by JukkaL** [comment]
If you looked up these from documentation, you should be aware of that (Python) documentation is not to be trusted as a source of truth when creating stubs -- docs tend to have too many inaccuracies and outright errors.

**Comment by z33ky** [comment]
Yeah I tried looking at the CPython code, but I didn't quite get what is happening there.

I checked for existence of the functions with the interpreter and removed the ones which failed.

The 4 comparison operators are taken out. A pull request for mypy, which would allow defining them, is here: python/mypy#2584.

**Comment by gvanrossum** [comment]
I've added `__format__` to the PY2 version as well, and removed the redundant `float.__format__`.

Though maybe it should have the argument type `AnyStr`? Or `Union[str, Text]`? I have to think about that.

There are also some redundant definitions of `__format__` in other modules now. Do you want to try and clean those up as well?

**Comment by gvanrossum** [comment]
@JukkaL can you approve and/or merge?

**Comment by JukkaL** [comment]
Thanks for the updates!

`__format__` should perhaps use `AnyStr` but I'll leave this to a future PR.

**Pull request merged**